### PR TITLE
e2e add one test as a sample

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

This PR adds one e2e test that can be used a sample of real tests.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,8 @@ require (
 	github.com/onsi/ginkgo/v2 v2.1.4
 	github.com/onsi/gomega v1.19.0
 	github.com/stretchr/testify v1.7.0
+	k8s.io/api v0.22.1
+	k8s.io/apiextensions-apiserver v0.22.1
```
Comment (timflannagan) [inline go.mod:11]: It looks like we'll need a 1.24 bump issue at some point.

**Review by timflannagan — APPROVED** [review]

File: test/e2e/deppy_suite_test.go [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
+	Expect(err).To(BeNil())
+
+	err = apiextensionsv1.AddToScheme(scheme)
+	Expect(err).To(BeNil())
+
+	c, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).To(BeNil())
+
+	kubeClient, err = kubernetes.NewForConfig(cfg)
+	Expect(err).To(BeNil())
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply test/e2e/deppy_suite_test.go:51]: I see.  I'll take it out.

File: test/e2e/deppy_suite_test.go [hunk 3]
```diff
@@ -1,13 +1,52 @@
...
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	deppyv1alpha1 "github.com/operator-framework/deppy/api/v1alpha1"
+)
+
+var (
+	cfg        *rest.Config
+	c          client.Client
+	kubeClient kubernetes.Interface
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply test/e2e/deppy_suite_test.go:24]: Yes, I'll take it out.

File: test/e2e/basic_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing Input resource")
+			input = &deppyv1alpha1.Input{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "input-valid",
+				},
+				Spec: deppyv1alpha1.InputSpec{
+					InputClassName: "input-class-name",
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply test/e2e/basic_test.go:38]: I'll do.  Thanks!

**Review by tylerslaton — APPROVED** [review]
Nice! This will serve as a good example moving forward.

**Comment by akihikokuroda** [comment]
@timflannagan @tylerslaton Thanks!  I don't have power to merge in this repo.  Would you merge this?

**Pull request merged**